extensions/standard-processors/tests/unit/TailFileTests.cpp

```diff
@@ -1536,3 +1514,197 @@ TEST_CASE("TailFile interprets the lookup frequency property correctly", "[multi
+
+    testController.runSession(plan);
+
+    REQUIRE(LogTestController::getInstance().contains("Logged 0 flow files"));
+
+    plan->reset(true);
+    LogTestController::getInstance().resetStream(LogTestController::getInstance().log_output);
+
+    appendTempFile(dir, TMP_FILE, NEW_TAIL_DATA);
+
+    testController.runSession(plan);
+
+    REQUIRE(LogTestController::getInstance().contains("Logged 1 flow files"));
+    REQUIRE(LogTestController::getInstance().contains("Size:" + std::to_string(NEW_TAIL_DATA.find_first_of('\n') + 1) + " Offset:0"));
+  }
+
+  LogTestController::getInstance().reset();
+}
+
+TEST_CASE("TailFile reads multiple files when Initial Start Position is set", "[initialStartPosition]") {
+  TestController testController;
+  LogTestController::getInstance().setTrace<minifi::processors::TailFile>();
+  LogTestController::getInstance().setDebug<minifi::processors::LogAttribute>();
+
+  std::shared_ptr<TestPlan> plan = testController.createPlan();
+  std::shared_ptr<core::Processor> tailfile = plan->addProcessor("TailFile", "tailfileProc");
+  std::shared_ptr<core::Processor> logattribute = plan->addProcessor("LogAttribute", "logattribute", core::Relationship("success", "description"), true);
+
+  auto dir = minifi::utils::createTempDir(&testController);
+  createTempFile(dir, ROLLED_OVER_TMP_FILE, ROLLED_OVER_TAIL_DATA);
+  createTempFile(dir, TMP_FILE, NEWLINE_FILE);
+  const std::string TMP_FILE_2_DATA = "tmp_file_2_new_line_data\n";
+  createTempFile(dir, "minifi-tmpfile-2.txt", TMP_FILE_2_DATA);
+
+  plan->setProperty(logattribute, org::apache::nifi::minifi::processors::LogAttribute::FlowFilesToLog.getName(), "0");
+  plan->setProperty(tailfile, org::apache::nifi::minifi::processors::TailFile::FileName.getName(), ".*\\.txt");
+  plan->setProperty(tailfile, org::apache::nifi::minifi::processors::TailFile::Delimiter.getName(), "\n");
+  plan->setProperty(tailfile, org::apache::nifi::minifi::processors::TailFile::TailMode.getName(), "Multiple file");
+  plan->setProperty(tailfile, org::apache::nifi::minifi::processors::TailFile::BaseDirectory.getName(), dir);
+
+  SECTION("Initial Start Position is set to Beginning of File") {
+    plan->setProperty(tailfile, org::apache::nifi::minifi::processors::TailFile::InitialStartPosition.getName(), "Beginning of File");
+
+    testController.runSession(plan);
+
+    REQUIRE(LogTestController::getInstance().contains("Logged 2 flow files"));
+    REQUIRE(LogTestController::getInstance().contains("Size:" + std::to_string(NEWLINE_FILE.find_first_of('\n') + 1) + " Offset:0"));
+    REQUIRE(LogTestController::getInstance().contains("Size:" + std::to_string(TMP_FILE_2_DATA.find_first_of('\n') + 1) + " Offset:0"));
+
+    plan->reset(true);
+    LogTestController::getInstance().resetStream(LogTestController::getInstance().log_output);
+
+    createTempFile(dir, "minifi-tmpfile-3.txt", NEWLINE_FILE);
```

Comment:
The test could be made clearer if the contents (and length) of `minifi-tmpfile-3.txt` were different from the contents (and length) of `TMP_FILE`.

Reply:
Updated in 221b1d76768daef020fe6d35be85dfad7aab2796